components/clp-package-utils/clp_package_utils/scripts/decompress.py

```diff
@@ -162,9 +166,11 @@ def handle_extract_stream_cmd(
         return -1
 
     storage_type = clp_config.archive_output.storage.type
-    if StorageType.FS != storage_type:
+    storage_engine = clp_config.package.storage_engine
+    if StorageType.S3 == storage_type and StorageEngine.CLP == storage_engine:
```

Comment:
nit: I might write 

if StorageType.S3 == storage_type and StorageEngine.CLP_S != storage_engine since we only expect CLP-s to support S3. but not a big deal